app/src/main/java/com/metrics_board/app/service/roll/ProjectService.java

```diff
@@ -60,4 +92,32 @@ private ProjectResponse createProjectResponse(Project project) {
                 .updatedAt(project.getUpdatedAt())
                 .build();
     }
+
+    private void validateProjectRequest(ProjectRequest request) {
+        if (request.getName() == null && request.getDescription() == null && request.getStatus() == null) {
+            throw new MissedDataToUpdateException("Missed data to update");
+        }
+
+        if (request.getName() != null && request.getName().equals("")) {
+            throw new MissedDataToUpdateException("Invalid name to update");
+        }
+
+        if (request.getStatus() != null && !request.getStatus().matches("^(active|suspended|archived)$")) {
+            throw new MissedDataToUpdateException("Invalid status to update");
+        }
+    }
+
+    private void updateProjectFields(ProjectRequest request, Project updateProject) {
+        if (request.getName() != null) {
+            updateProject.setName(request.getName());
+        }
+
+        if (request.getDescription() != null) {
+            updateProject.setDescription(request.getDescription().equals("") ? null : request.getDescription());
+        }
+
+        if (request.getStatus() != null) {

```

Comment:
Here you performing double check one in `if` statement and second in the `ProjectStatus.of()` it can be safely done

`ProjectStatus.of(request.getStatus()).ifPresent(updateProject::setStatus)`